### ISSUE TITLE
failed to create model: Replicate API error: Unknown error: json: cannot unmarshal object into Go struct field APIError.detail of type string

### ISSUE THREAD

**Issue by shripadk** [replicate/cli#41]
I get the following error when trying to create a model via the CLI tool. Any idea what could be the issue here? I have set the API Key correctly and `replicate model show <owner-redacted>/<name>` works:

```bash
» replicate model create <owner-redacted>/<name-redacted>  --private --hardware gpu-a40-large
Error: failed to create model: failed to create model: Replicate API error: Unknown error: json: cannot unmarshal object into Go struct field APIError.detail of type string
Usage:
  replicate model create <owner>/<name> [flags]

Flags:
      --cover-image-url string   URL of the cover image
      --description string       Description of the model
      --github-url string        URL of the GitHub repository
      --hardware string          SKU of the hardware to run the model
  -h, --help                     help for create
      --json                     Emit JSON
      --license-url string       URL of the license
      --paper-url string         URL of the paper
      --private                  Make the new model private
      --public                   Make the new model public
      --web                      View on web
```

**Comment by mattt** [comment]
Hi @shripadk. This is a problem on our side. Our API is sending an error in an unexpected format, and the CLI is returning an error about that. Could you try making the equivalent call via `curl` to see what the API is sending back?

```console
$ curl -s -X POST -H "Authorization: Token $REPLICATE_API_TOKEN" \
    -d '{"owner": "my-username", "name": "my-new-model", "visibility": "private", "hardware": "gpu-a40-large"}' \
    https://api.replicate.com/v1/models 
```

**Comment by shripadk** [comment]
Hi @mattt thanks for the quick response. Sure. I get the following error response:

```bash
» echo $REPLICATE_API_TOKEN
r8_******************************

» curl -s -X POST -H "Authorization: Token $REPLICATE_API_TOKEN" \
          -d '{"owner": "<redacted>", "name": "this-is-a-test-model", "visibility": "private", "hardware": "gpu-a40-large"}' \
          https://api.replicate.com/v1/models
{"detail":{"hardware":"This field is required.","name":"This field is required.","owner":"This field is required."}}⏎
```

**Comment by mattt** [comment]
@shripadk Thank you! That's extremely helpful. I know exactly what's wrong and how to fix it. Hoping to have an update by the end of the week.

**Comment by mattt** [comment]
@shripadk Thanks for your patience. I figured out the problem and have a pending fix with https://github.com/replicate/cli/pull/46.

**Comment by mattt** [comment]
@shripadk This is now fixed in [version 0.5.1](https://github.com/replicate/cli/releases/tag/v0.5.1). Thanks again for your help!

**Comment by shripadk** [comment]
Thanks @mattt! Just got to see the fix. Can confirm it works! Thanks again!
